submitit/core/core.py

```diff
@@ -594,12 +568,12 @@ def batch(self) -> tp.Iterator[None]:
         jobs, submissions = zip(*delayed_batch)
         new_jobs = self._internal_process_submissions(submissions)
         for j, new_j in zip(jobs, new_jobs):
-            j._mutate(new_j)
+            j.__dict__.update(new_j.__dict__)  # fill in the empty shell, the pickle way
 
     def submit(self, fn: tp.Callable[..., R], *args: tp.Any, **kwargs: tp.Any) -> Job[R]:
         ds = utils.DelayedSubmission(fn, *args, **kwargs)
         if self._delayed_batch is not None:
-            job: Job[R] = JobPlaceholder()
+            job: Job[R] = self.job_class.__new__(self.job_class)  # empty shell
```

Comment:
Here I feel we should try to fill more of the jobs properties: the folder_path, tasks, ...

Reply:
After discussion with Jérémy, I realized that since the paths can't be resolved without the job id, there is not much point to add the paths early.